src/main/resources/assets/admin/common/js/ui/selector/Option.ts

```diff
@@ -1,5 +1,82 @@
-export interface Option<T>
-    extends Slick.SlickData {
+import {IDentifiable} from '../../IDentifiable';
+
+export class Option<T>
+    implements Slick.SlickData, IDentifiable {
+
+    private value: string;
+
+    private displayValue: T;
+
+    private indices?: string[];
+
+    private readOnly?: boolean;
+
+    private empty?: boolean;
+
+    private selectable?: boolean;
+
+    private expandable?: boolean;
+
+    constructor(builder: OptionBuilder<T>) {
+        this.value = builder.value;
+        this.displayValue = builder.displayValue;
+        this.indices = builder.indices;
+        this.readOnly = builder.readOnly;
+        this.empty = builder.empty;
+        this.selectable = builder.selectable;
+        this.expandable = builder.expandable;
+    }
+
+    setValue(value: string) {
+        this.value = value;
+    }
+
+    getValue(): string {
+        return this.value;
+    }
+
+    setDisplayValue(value: T) {
+        this.displayValue = value;
+    }
+
+    getDisplayValue(): T {
+        return this.displayValue;
+    }
+
+    getIndices(): string[] {
+        return this.indices;
+    }
+
+    setReadOnly(value: boolean) {
+        this.readOnly = value;
+    }
+
+    isReadOnly(): boolean {
+        return this.readOnly;
+    }
+
+    isEmpty(): boolean {
+        return this.empty;
+    }
+
+    isSelectable(): boolean {
+        return this.selectable;
+    }
+
+    isExpandable(): boolean {
+        return this.expandable;
+    }
+
+    static create<T>(): OptionBuilder<T> {
+        return new OptionBuilder<T>();
```

Comment:
Codacy found an issue: ['OptionBuilder' was used before it was defined.](https://app.codacy.com/manual/enonic/lib-admin-ui/pullRequest?prid=6401650)